### PR TITLE
Associate view

### PR DESCRIPTION
**Pull request by hannahnov**
Created survey model, DTO, service, and controller. Created testing for basic CRUD methods in the survey service

### CODE REVIEW

File: SyncService/src/main/java/com/revature/controllers/SurveyController.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import com.revature.models.Survey;
+import com.revature.service.SurveyService;
+
```
Comment (jurczakn) [inline SyncService/src/main/java/com/revature/controllers/SurveyController.java:20]: add comments

File: SyncService/src/main/java/com/revature/models/Survey.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
+package com.revature.models;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+/**
+ * 
+ * @author 
+ * Hannah
```
Comment (jurczakn) [inline SyncService/src/main/java/com/revature/models/Survey.java]: format

File: SyncService/src/main/java/com/revature/service/SurveyServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package com.revature.service;
+
+import com.revature.models.Survey;
+import com.revature.repo.SurveyRepo;
+
```
Comment (jurczakn) [inline SyncService/src/main/java/com/revature/service/SurveyServiceImpl.java:5]: no documentation

File: SyncService/src/test/java/com/revature/controllers/SurveyControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.revature.dto.SurveyDto;
+import com.revature.models.QuestionType;
+import com.revature.models.Survey;
+import com.revature.models.SurveyQuestion;
+import com.revature.service.SurveyService;
+
+/**
+ * @author Hannah Brett Alma Marc Yarashlee
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/controllers/SurveyControllerTest.java:36]: more documentation

File: SyncService/src/test/java/com/revature/controllers/SurveyControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+	/**
+	 * @throws java.lang.Exception
+	 */
+	@AfterEach
+	void tearDown() throws Exception {
+	}
+
+	@Test
+	void getSurveyByIdTest() {
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/controllers/SurveyControllerTest.java]: not sure what trying to test, make more clear in test name and implementation

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+import com.revature.repo.SurveyRepo;
+
+
+@SpringBootTest()
+@AutoConfigureMockMvc
+@RunWith(SpringRunner.class)
+public class SurveyServiceTest {
+	
+	@Mock
+
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:36]: fix spacing

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+	@Test
+	void surveyServiceTest_WithoutError() {
+		
+		when(surveyRepo.createSurvey(survey)).thenReturn(survey);
+		
+		Survey returned = surveyService.createSurvey(survey);
+		
+		verify(surveyRepo).createSurvey(survey);
+		
+		assertEquals(survey, returned, "Object returned does not match expected.");
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:79]: same, write better msg

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+		
+	}
+	
+	/*
+	 * Happy path for getting a survey
+	 * Can't retrieve survey because it doesn't exist
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:85]: not testing can't retrieve, testing can retrieve

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+		
+	}
+	
+	/**
+	 * Happy path, creates survey
+	 * @author Hannah, Alma, Brett, and Yara
+	 */
+	
+	@Test
+	void surveyServiceTest_WithoutError() {
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:71]: make name more specific

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+	 * Can't retrieve survey because it doesn't exist
+	 * @author Hannah, Alma, Brett, and Yara
+	 */
+	
+	@Test
+	void surveyServiceTest_SurveyDoesntExist() throws NullPointerException {
+		
+		when(surveyRepo.getSurvey(survey.getId())).thenReturn(null);
+		
+		assertThrows(NullPointerException.class, () -> surveyService.getSurvey(survey.getId()));
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:112]: should be throwing more specific exception

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+	}
+	
+	
+	/**
+	 * Edge case #2
+	 * Cannot create survey because it is null
+	 * @author Hannah, Alma, Brett, and Yara
+	 */
+	@Test
+	void surveyServiceTest_CannotCreateSurvey() {
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:122]: make name more specific

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+		
+		survey = null;
+		
+		when(surveyRepo.createSurvey(survey)).thenReturn(null);
+		
+		assertThrows(NullPointerException.class, () -> surveyService.createSurvey(survey));
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:128]: more specific exception

File: SyncService/src/main/java/com/revature/service/SurveyService.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+	 * @return 
+	 */
+	
+	public Survey getSurvey(int id);
+	
+	public Survey createSurvey(Survey survey);
+	
+	public Survey updateSurvey(Survey survey);
+	
+	public int deleteSurvey(Survey survey);
```
Comment (jurczakn) [inline SyncService/src/main/java/com/revature/service/SurveyService.java]: return something more useful

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+	}
+	
+	/**
+	 * Edge case #3
+	 * Cannot delete survey because it doesn't exist
+	 * @author Hannah, Alma, Brett, and Yara
+	 */
+	
+	@Test
+	void surveyServiceTest_CannotDeleteSurvey() {
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:156]: why cannot delete survey, make name clearer

File: SyncService/src/test/java/com/revature/service/SurveyServiceTest.java [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+		
+
```
Comment (jurczakn) [inline SyncService/src/test/java/com/revature/service/SurveyServiceTest.java:158]: spacing

**Review by jurczakn — CHANGES_REQUESTED** [review]
